### ISSUE TITLE
[Bug]: Sidebar item hover background colour makes text unreadable

### ISSUE THREAD

**Issue by NickyBlissTui** [storybookjs/storybook#30284]
### Describe the bug

The sidebar item hover background colour variable "--tree-node-background-hover" is being updated on hover and focus to a lighter or darker shade of the themes secondary colour but the font colour is not being updated to accommodate this change mean that now the text is unreadable when the items are in hover or focus.

![Image](https://github.com/user-attachments/assets/d6eb82ce-77df-475e-87e7-3dbf06f57ac8)

### Reproduction link

https://stackblitz.com/edit/github-djnsqscp?file=package.json

### Reproduction steps

1. Update from 8.4.7 to 8.5.0+
2. Set colorSecondary for the storybook theme to a dark colour eg. #1a115a
3. Run storybook
4. hover/focus on unselected items within the sidemenu are now almost unreadable

### System

```bash
Storybook Environment Info:

  System:
    OS: macOS 15.2
    CPU: (12) arm64 Apple M3 Pro
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 20.18.0 - /usr/local/bin/node
    npm: 10.8.2 - /usr/local/bin/npm <----- active
  Browsers:
    Chrome: 132.0.6834.84
    Edge: 131.0.2903.147
    Safari: 18.2
  npmPackages:
    @storybook/addon-a11y: ^8.4.7 => 8.5.0 
    @storybook/addon-docs: ^8.4.7 => 8.5.0 
    @storybook/addon-essentials: ^8.4.7 => 8.5.0 
    @storybook/addon-links: ^8.4.7 => 8.5.0 
    @storybook/addon-themes: ^8.4.7 => 8.5.0 
    @storybook/blocks: ^8.4.7 => 8.5.0 
    @storybook/html: ^8.4.7 => 8.5.0 
    @storybook/html-webpack5: ^8.4.7 => 8.5.0 
    @storybook/manager-api: ^8.4.7 => 8.5.0 
    @storybook/preview-api: ^8.4.7 => 8.5.0 
    @storybook/theming: ^8.4.7 => 8.5.0 
    storybook: ^8.4.7 => 8.5.0 
    storybook-addon-pseudo-states: ^4.0.2 => 4.0.2 
    storybook-addon-tag-badges: ^1.4.0 => 1.4.0 
    storybook-addon-turbo-build: ^2.0.1 => 2.0.1 
    storybook-dark-mode: ^4.0.2 => 4.0.2 
    storybook-preset-inline-svg: ^1.0.1 => 1.0.1
```

**Comment by msmx-mnakagawa** [comment]
(cc: @NickyBlissTui)
I have tried to make a PR fixing this issue.
#30287

If possible, could someone take a look into the PR?

**Comment by michaelokolo** [comment]
Hi @NickyBlissTui,
I hope you're well. Is this issue still open? If so, may I kindly work on it?

Thank you!

**Comment by NickyBlissTui** [comment]
@michaelokolo There is a PR with a potential fix submitted but would be good if it was double checked to make sure it fixes this issue but also does not cause any other issues with theming.

**Comment by NickyBlissTui** [comment]
> Hi [@NickyBlissTui](https://github.com/NickyBlissTui) , I'm interested in contributing to this issue. Could you please assign it to me? Looking forward to working on it!

I only raised the issue I dont have permission to do anything more with this ticket

**Comment by Sidnioulz** [comment]
@MichaelArestad tempted to close this as duplicate of #8935. Waiting for you to confirm if you agree :) 

Folks, this issue arises from problems in the vocabulary used in themes. If we fixed the contrast problem for the particular theme values in this issue, we might just reintroduce it for other theme values. The underlying problem is that the variable names in the theme API don't allow you to predict which colours will be used where.

There are two ongoing projects to address that:
* https://github.com/storybookjs/storybook/discussions/24344
* https://github.com/storybookjs/storybook/discussions/32013

**Comment by MichaelArestad** [comment]
Those two projects will address it as we will be able to be more rigorous about color transforms. I might keep this around if possible just so we ensure this isn't overlooked. I don't think the changes I'm making elsewhere address this.

**Comment by Sidnioulz** [comment]
Thanks for clarifying @MichaelArestad! I'll keep the issue open till then.

**Comment by Sidnioulz** [comment]
@MichaelArestad can we close this issue? It's fixed, isn't it?

**Comment by MichaelArestad** [comment]
This should be fixed!
